### PR TITLE
Add mujoco sawyer envs

### PR DESCRIPTION
**Pull request by zhanpenghe**
Three environments are added for the sawyer.
- PickAndPlace
- BinSorting
- BlockStacking

For now only sparse reward are implemented and we need to explore for a better reward function.

Address #21

### CODE REVIEW

File: garage/envs/mujoco/sawyer/bin_sorting_env.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import numpy as np
+
+from garage.core.serializable import Serializable
```
Comment (ryanjulian) [inline garage/envs/mujoco/sawyer/bin_sorting_env.py]: use short imports

`from garage.core import Serializable`, etc.

File: garage/envs/mujoco/sawyer/block_stacking_env.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+from garage.core.serializable import Serializable
```
Comment (ryanjulian) [inline garage/envs/mujoco/sawyer/block_stacking_env.py]: short imports

File: garage/envs/mujoco/sawyer/pick_and_place_env.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+from gym.spaces import Box
+import numpy as np
+
+from garage.core.serializable import Serializable
+from garage.envs.base import Step
+from garage.envs.mujoco.mujoco_env import MujocoEnv
+from garage.misc.overrides import overrides
+
+
+def goal_distance(goal_a, goal_b):
```
Comment (ryanjulian) [inline garage/envs/mujoco/sawyer/pick_and_place_env.py]: please make this private `def _goal_distance(goal_a, goal_b)` or make it a static method 

```python
class PickAndPlaceEnv():

    @staticmethod
    def _goal_distance(goal_a, goal_b):
        # stuff
```

File: tests/envs/test_sawyer_envs.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import numpy as np
+
+from garage.algos.trpo import TRPO
```
Comment (ryanjulian) [inline tests/envs/test_sawyer_envs.py]: use short imports

File: tests/envs/test_sawyer_envs.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+from garage.baselines import LinearFeatureBaseline
+from garage.envs.mujoco.sawyer import BinSortingEnv, BlockStackingEnv, PickAndPlaceEnv
+from garage.envs.util import spec
+from garage.misc.instrument import run_experiment_lite
+from garage.policies.gaussian_mlp_policy import GaussianMLPPolicy
```
Comment (ryanjulian) [inline tests/envs/test_sawyer_envs.py]: use short imports

File: garage/envs/mujoco/sawyer/pick_and_place_env.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
+from gym.envs.robotics import rotations
+from gym.spaces import Box
+import numpy as np
+
+from garage.core.serializable import Serializable
+from garage.envs.base import Step
+from garage.envs.mujoco.mujoco_env import MujocoEnv
```
Comment (ryanjulian) [inline garage/envs/mujoco/sawyer/pick_and_place_env.py]: short imports

**Comment by ryanjulian** [comment]
LGTM

Please file an issue to tune these envs to match the actual USC sawyer setup.

**Comment by eric-heiden** [comment]
I think it would be nice to have these environments be a [GoalEnv](https://github.com/openai/gym/blob/5404b39d06f72012f562ec41f60734bd4b5ceb4b/gym/core.py#L154) so that we can expose the function to compute rewards, similar to the Gym [`robotics` environments](https://github.com/openai/gym/tree/master/gym/envs/robotics) which nicely integrate with HER-style algorithms. To that end, a simple 3d point reacher environment would also be a good contribution.

**Comment by ryanjulian** [comment]
@eric-heiden file an issue?

**Comment by zhanpenghe** [comment]
@ryanjulian @eric-heiden Issue filed.

File: garage/envs/mujoco/sawyer/bin_sorting_env.py (hunk 1 above)
Comment (zhanpenghe) [reply garage/envs/mujoco/sawyer/bin_sorting_env.py]: Done.

File: garage/envs/mujoco/sawyer/pick_and_place_env.py (hunk 3 above)
Comment (zhanpenghe) [reply garage/envs/mujoco/sawyer/pick_and_place_env.py]: Done.

**Pull request merged**